services/osparc-gateway-server/src/osparc_gateway_server/backend/osparc.py

```diff
@@ -1,298 +1,221 @@
-            "RestartPolicy": {"Condition": "on-failure"},
-        },
-        "networks": [network_id],
-        # "mode": {"Global": {}},
-    }
 
-
-class OsparcClusterConfig(ClusterConfig):
-    """Dask cluster configuration options when running as osparc backend"""
-
-
-class OsparcBackend(LocalBackend):
+class OsparcBackend(DBBackendBase):
     """A cluster backend that launches osparc workers.
 
     Workers are spawned as services in a docker swarm
     """
 
-    cluster_config_class = Type(
-        "osparc_gateway_server.backend.osparc.OsparcClusterConfig",
-        klass="dask_gateway_server.backends.base.ClusterConfig",
-        help="The cluster config class to use",
-        config=True,
-    )
-
     default_host = "0.0.0.0"
     # worker_start_timeout = 120
 
     settings: AppSettings
+    docker_client: Docker
+    cluster_secrets: List[DockerSecret] = []
+
+    clusters_directory = Unicode(
+        "/tmp/clusters_directory",
+        help="Path to use for keeping the clusters directories",
+        config=True,
+    )
 
     async def do_setup(self) -> None:
         await super().do_setup()
         self.settings = AppSettings()
+        self.docker_client = Docker()
         self.log.info(
             "osparc-gateway-server application settings:\n%s",
             self.settings.json(indent=2),
         )
 
-    async def do_start_worker(
-        self, worker: Worker
-    ) -> AsyncGenerator[Dict[str, Any], None]:
-        self.log.debug("received call to start worker as %s", f"{worker}")
-
-        scheduler_url = urlsplit(worker.cluster.scheduler_address)
-        # scheduler_host = scheduler_url.netloc.split(":")[0]
-        port = scheduler_url.netloc.split(":")[1]
-        netloc = f"{self.settings.GATEWAY_SERVER_NAME}:{port}"
-        scheduler_address = urlunsplit(scheduler_url._replace(netloc=netloc))
-
-        # db_address = f"{self.default_host}:8787"
-        workdir = worker.cluster.state.get("workdir")
+    async def do_cleanup(self) -> None:
+        await super().do_cleanup()
+        await self.docker_client.close()
 
-        # nthreads, memory_limit = self.worker_nthreads_memory_limit_args(worker.cluster)
-
-        workdir = worker.cluster.state.get("workdir")
-        self.log.debug("workdir set as %s", f"{workdir}")
+    async def do_start_cluster(
+        self, cluster: Cluster
+    ) -> AsyncGenerator[Dict[str, Any], None]:
+        self.log.debug(f"starting cluster {cluster=}")
+        self.cluster_secrets.extend(
+            await _create_docker_secrets_from_tls_certs_for_cluster(
+                self.docker_client, self, cluster
+            )
+        )
+        self.log.debug(
+            "created secrets for TLS certification: %s", f"{self.cluster_secrets=}"
+        )
 
-        service_parameters = None
+        # now we need a scheduler
+        scheduler_env = self.get_scheduler_env(cluster)
+        scheduler_cmd = self.get_scheduler_command(cluster)
```

Comment:
So you take them from the base class as is?

Reply:
yes, with some small modifications regarding the hostname of the dask-gateway.